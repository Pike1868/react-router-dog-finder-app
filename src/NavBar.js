import { NavLink } from "react-router-dom";

/**
 *
 * NavBar Component:
 *
 * Returns a navigation bar with each dog object listed as a NavLink component. allowing
 * users to click on a dog's name and navigate to the route displaying that dog's details.
 * Also includes a link to the home route ("/dogs") that directs users back
 * to the main list of all dogs.
 *
 * Props:
 * - dogs (array): An array of objects representing the dogs, used to generate navigation links.
 *
 */

function NavBar({ dogs }) {
  const links = dogs.map((dog) => (
    <NavLink key={dog.name} to={`/dogs/${dog.name.toLowerCase()}`}>
      {dog.name}
    </NavLink>
  ));

  return (
    <nav
      style={{
        display: "flex",
        justifyContent: "space-around",
        backgroundColor: "lightblue",
        padding: "10px",
      }}
    >
      <NavLink to="/dogs" end>
        Home
      </NavLink>
      {links}
    </nav>
  );
}

export default NavBar;
