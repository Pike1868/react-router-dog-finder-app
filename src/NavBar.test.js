import React from "react";
import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import NavBar from "./NavBar";
import dogs from "./dogs";

//Smoke Test
test("NavBar component renders without crashing", () => {
  render(
    <MemoryRouter>
      <NavBar dogs={dogs} />
    </MemoryRouter>
  );
});

//Snapshot Test
test("NavBar component matches snapshot", () => {
  const { asFragment } = render(
    <MemoryRouter>
      <NavBar dogs={dogs} />
    </MemoryRouter>
  );

  expect(asFragment()).toMatchSnapshot();
});

describe("NavBar Component Tests", () => {
  beforeEach(() => {
    render(
      <MemoryRouter>
        <NavBar dogs={dogs} />
      </MemoryRouter>
    );
  });

  test("displays the home link", () => {
    const homeLink = screen.getByText("Home");
    expect(homeLink).toBeInTheDocument();
  });

  test("displays a dog link", () => {
    const whiskeyLink = screen.getByText("Whiskey");
    expect(whiskeyLink).toBeInTheDocument();
  });
});
