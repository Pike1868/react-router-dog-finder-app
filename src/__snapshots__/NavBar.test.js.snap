// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar component matches snapshot 1`] = `
<DocumentFragment>
  <nav
    style="display: flex; justify-content: space-around; background-color: lightblue; padding: 10px;"
  >
    <a
      class=""
      href="/dogs"
    >
      Home
    </a>
    <a
      class=""
      href="/dogs/whiskey"
    >
      Whiskey
    </a>
    <a
      class=""
      href="/dogs/duke"
    >
      Duke
    </a>
    <a
      class=""
      href="/dogs/perry"
    >
      Perry
    </a>
    <a
      class=""
      href="/dogs/tubby"
    >
      Tubby
    </a>
  </nav>
</DocumentFragment>
`;
